scalanet/test/src/io/iohk/scalanet/dynamictls/CustomTlsValidatorSpec.scala

```diff
@@ -0,0 +1,120 @@
+package io.iohk.scalanet.dynamictls
+
+import java.security.SecureRandom
+import java.time.{Clock, LocalDateTime, ZoneOffset}
+import java.util.Date
+
+import io.iohk.scalanet.crypto.CryptoUtils
+import io.iohk.scalanet.crypto.CryptoUtils.Secp256r1
+import io.iohk.scalanet.peergroup.dynamictls.CustomTlsValidator.{
+  NoCertExtension,
+  SeverIdNotMatchExpected,
+  WrongCertificateDate,
+  WrongExtensionFormat,
+  WrongExtensionSignature,
+  WrongNumberOfCertificates
+}
+import io.iohk.scalanet.peergroup.dynamictls.DynamicTLSExtension.{Extension, SignedKey, SignedKeyExtensionNodeData}
+import io.iohk.scalanet.peergroup.dynamictls.{CustomTlsValidator, Secp256k1}
+import io.iohk.scalanet.testutils.GeneratorUtils
+import org.bouncycastle.asn1.DERGeneralString
+import org.scalatest.prop.GeneratorDrivenPropertyChecks.forAll
+import org.scalatest.prop.GeneratorDrivenPropertyChecks._
+import org.scalatest.{FlatSpec, Matchers}
+import scodec.bits.BitVector
+
+class CustomTlsValidatorSpec extends FlatSpec with Matchers {
+  val rnd = new SecureRandom()
+  val today = LocalDateTime.now(Clock.systemUTC())
+  val beforeDate = Date.from(today.minusMonths(1).toInstant(ZoneOffset.UTC))
+  val afterDate = Date.from(today.plusYears(100).toInstant(ZoneOffset.UTC))
```

Comment:
as it looks like some const ways to produce before and after date, maybe it would be good to use some factory to get them ?

Reply:
i think you are right this manual conversion everywhere looks shity. I will add https://github.com/nscala-time/nscala-time library, which supports `Interval` class, and have implemented `toDate` conversions. (unfortunatly `X509Certs` require old java `Date` class as args)